planemo/commands/cmd_test.py

```diff
@@ -66,7 +68,11 @@ def cli(ctx, uris, **kwds):
     against that same Galaxy root - but this may not be bullet proof yet so
     please careful and do not try this against production Galaxy instances.
     """
-    with temp_directory(dir=ctx.planemo_directory) as temp_path:
+    with ExitStack() as stack:
+        if not kwds["serve"]:
+            temp_path = stack.enter_context(temp_directory(dir=ctx.planemo_directory))
```

Comment:
Fascinating! I had no clue about this.

Reply:
Yes, new for me as well, I came across it while searching for a way to write a conditional with statement.